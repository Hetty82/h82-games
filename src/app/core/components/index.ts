import { AppComponent } from './app/app.component'
import { LayoutComponent } from './layout/layout.component'
import { NotFoundComponent } from './not-found/not-found.component'

export const components: any[] = [
  AppComponent,
  LayoutComponent,
  NotFoundComponent,
]

export * from './app/app.component'
export * from './layout/layout.component'
export * from './not-found/not-found.component'
