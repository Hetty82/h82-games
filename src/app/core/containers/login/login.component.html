<section class="container">
  <h1 class="page-title">Choose player:</h1>

  <ul class="list">
    <li *ngFor="let user of (users$ | async)" (click)="selectPlayer(user.id)" class="item">
      <button class="button">{{ user.name }}</button>
    </li>
  </ul>
</section>
