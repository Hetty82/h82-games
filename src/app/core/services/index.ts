import { GamesService } from './games.service'
import { UserService } from './user.service'

export const services: any[] = [
  GamesService,
  UserService,
]

export * from './games.service'
export * from './user.service'
