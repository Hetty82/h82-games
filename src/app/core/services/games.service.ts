import { Injectable } from '@angular/core'
import { HttpClient } from '@angular/common/http'

import { Observable } from 'rxjs/Observable'

import { environment } from '../../../environments/environment'

import { Game } from '../models/game.interface'


@Injectable()
export class GamesService {
  constructor(
    private http: HttpClient,
  ) { }

  getGames() {
    const url = environment.api.games

    return this.http.get(url) as Observable<Game[]>
  }
}
