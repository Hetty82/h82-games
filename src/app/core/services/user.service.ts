import { Injectable } from '@angular/core'
import { HttpClient } from '@angular/common/http'

import { Observable } from 'rxjs/Observable'

import { environment } from '../../../environments/environment'

import { User } from '../models/user.interface'


@Injectable()
export class UserService {
  constructor(
    private http: HttpClient
  ) { }

  getUsers() {
    const url = environment.api.users
    return this.http.get(url) as Observable<User[]>
  }
}
