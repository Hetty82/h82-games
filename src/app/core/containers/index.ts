import { AppComponent } from './app/app.component'
import { GamesComponent } from './games/games.component'
import { LoginComponent } from './login/login.component'
import { NotFoundComponent } from './not-found/not-found.component'

export const containers: any[] = [
  AppComponent,
  GamesComponent,
  LoginComponent,
  NotFoundComponent,
]

export * from './app/app.component'
export * from './games/games.component'
export * from './login/login.component'
export * from './not-found/not-found.component'
