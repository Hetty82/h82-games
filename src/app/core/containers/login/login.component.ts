import { Component, OnInit } from '@angular/core'
import { Observable } from 'rxjs/Observable'

import { UserService } from '../../services'
import { User } from '../../interfaces/user.interface'


@Component({
  selector: 'app-login',
  templateUrl: './login.component.html',
  styleUrls: ['./login.component.sass']
})
export class LoginComponent implements OnInit {
  users$: Observable<User[]> = this.userService.getUsers()

  constructor(
    private userService: UserService
  ) { }

  ngOnInit() {
  }
}
