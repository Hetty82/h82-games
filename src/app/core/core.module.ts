import { NgModule } from '@angular/core'
import { CommonModule } from '@angular/common'
import { RouterModule } from '@angular/router'

import * as fromComponents from './components'
import * as fromContainers from './containers'
import * as fromServices from './services'


@NgModule({
  imports: [
    CommonModule,
    RouterModule,
  ],
  declarations: [
    ...fromComponents.components,
    ...fromContainers.containers,
  ],
  exports: [
    fromContainers.AppComponent,
  ],
})
export class CoreModule {
  static forRoot() {
    return {
      ngModule: CoreModule,
      providers: fromServices.services,
    }
  }
}
