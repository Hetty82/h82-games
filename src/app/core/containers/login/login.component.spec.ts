import { NO_ERRORS_SCHEMA } from '@angular/core'
import { TestBed, async, ComponentFixture } from '@angular/core/testing'

import { UserService } from '../../services'
import { LoginComponent } from './login.component'


describe('LoginComponent', () => {
  let component: LoginComponent
  let fixture: ComponentFixture<LoginComponent>

  beforeEach(async(() => {
    TestBed.configureCompiler({ preserveWhitespaces: false } as any).configureTestingModule({
      declarations: [ LoginComponent ],
      providers: [
        {
          provide: UserService,
          useValue: {
            getUsers: jest.fn()
          }
        }
      ],
      schemas: [ NO_ERRORS_SCHEMA ],
    })
    .compileComponents()
  }))

  beforeEach(() => {
    fixture = TestBed.createComponent(LoginComponent)
    component = fixture.componentInstance
    fixture.detectChanges()
  })

  it('should create component', () => {
    expect(component).toBeTruthy()
  })

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot()
  })
})
