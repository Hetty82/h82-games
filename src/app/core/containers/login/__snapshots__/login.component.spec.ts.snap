// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginComponent should match snapshot 1`] = `
<app-login
  userService={[Function Object]}
  users$="undefined"
>
  <h2>
    User list
  </h2><ul>
    
  </ul>
</app-login>
`;
